go-controller/pkg/node/gateway_init.go

```diff
@@ -207,7 +209,7 @@ func (n *OvnNode) initGateway(subnets []*net.IPNet, nodeAnnotator kube.Annotator
 		gw, err = newLocalGateway(n.name, subnets, gatewayNextHops, gatewayIntf, nodeAnnotator, n.recorder, managementPortConfig)
 	case config.GatewayModeShared:
 		klog.Info("Preparing Shared Gateway")
-		gw, err = newSharedGateway(n.name, subnets, gatewayNextHops, gatewayIntf, egressGWInterface, nodeAnnotator)
+		gw, err = newSharedGateway(n.name, subnets, gatewayNextHops, gatewayIntf, egressGWInterface, nodeAnnotator, managementPortConfig, n.watchFactory.(*factory.WatchFactory))
```

Comment:
@trozet I passed n.watchFactory.(*factory.WatchFactory) so I can get direct Access to the methods exposed by https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/factory/types.go#L16, this is also what we do here https://github.com/astoycos/ovn-kubernetes/blob/externaltrafficpol/go-controller/pkg/node/node.go#L468

Reply:
Don't cast this, add the needed methods to the NodeWatchFactory interface.

Reply:
Fixing